lib/ProductOpener/FoodGroups.pm

```diff
@@ -0,0 +1,437 @@
+				}
+			}
+		}
+		else {
+			if ((not has_tag($product_ref, "categories", "en:non-alcoholic-beverages"))
+				and (not has_tag($product_ref, "categories", "en:alcoholic-beverages")) ) {
+				add_tag($product_ref, "categories", "en:non-alcoholic-beverages");
+			}
+		}
+
+		if ((not (has_tag($product_ref,"categories","en:alcoholic-beverages"))
+			or has_tag($product_ref,"categories","en:fruit-juices")
+			or has_tag($product_ref,"categories","en:fruit-nectars") ) ) {
+
+			if (has_tag($product_ref,"categories","en:sodas") and (not has_tag($product_ref,"categories","en:diet-sodas"))) {
+				if (not has_tag($product_ref,"categories","en:sweetened-beverages"))  {
+					add_tag($product_ref, "categories", "en:sweetened-beverages");
+				}
+				if (has_tag($product_ref,"categories","en:unsweetened-beverages")) {
+					remove_tag($product_ref, "categories", "en:unsweetened-beverages");
+				}
+			}
+
+			if ($product_ref->{with_sweeteners}) {
+				if (not has_tag($product_ref,"categories","en:artificially-sweetened-beverages")) {
+					add_tag($product_ref, "categories", "en:artificially-sweetened-beverages");
+				}
+				if (has_tag($product_ref,"categories","en:unsweetened-beverages")) {
+					remove_tag($product_ref, "categories", "en:unsweetened-beverages");
+				}
+			}
+			# fix me: ingredients are now partly taxonomized
+
+			if (
+
+				(has_tag($product_ref, "ingredients", "sucre") or has_tag($product_ref, "ingredients", "sucre-de-canne")
+				or has_tag($product_ref, "ingredients", "sucre-de-canne-roux") or has_tag($product_ref, "ingredients", "sucre-caramelise")
+				or has_tag($product_ref, "ingredients", "sucre-de-canne-bio") or has_tag($product_ref, "ingredients", "sucres")
+				or has_tag($product_ref, "ingredients", "pur-sucre-de-canne") or has_tag($product_ref, "ingredients", "sirop-de-sucre-inverti")
+				or has_tag($product_ref, "ingredients", "sirop-de-sucre-de-canne") or has_tag($product_ref, "ingredients", "sucre-bio")
+				or has_tag($product_ref, "ingredients", "sucre-de-canne-liquide") or has_tag($product_ref, "ingredients", "sucre-de-betterave")
+				or has_tag($product_ref, "ingredients", "sucre-inverti") or has_tag($product_ref, "ingredients", "canne-sucre")
+				or has_tag($product_ref, "ingredients", "sucre-glucose-fructose") or has_tag($product_ref, "ingredients", "glucose-fructose-et-ou-sucre")
+				or has_tag($product_ref, "ingredients", "sirop-de-glucose") or has_tag($product_ref, "ingredients", "glucose")
+				or has_tag($product_ref, "ingredients", "sirop-de-fructose") or has_tag($product_ref, "ingredients", "saccharose")
+				or has_tag($product_ref, "ingredients", "sirop-de-fructose-glucose") or has_tag($product_ref, "ingredients", "sirop-de-glucose-fructose-de-ble-et-ou-de-mais")
+				or has_tag($product_ref, "ingredients", "sugar") or has_tag($product_ref, "ingredients", "sugars")
+				or has_tag($product_ref, "ingredients", "en:sugar")
```

Comment:
Putting only one `has_tag` by line, seems more readable.

Really, I would use  `List::Util ``any`:
```perl
my @sweeteners = qw(sucre sucre-de-canne-roux sucre-caramelise ... )
if (any { has_tag($product_ref", "ingredients", $_)  } @sweeteners )
```

Reply:
Good idea, but same here, we will soon replace all that code because we will have a single en:added-sugar parent.

Reply:
yes ok, so leave it that way :-)